documentation/Using-the-SDK/Using-the-engine.md

```diff
@@ -0,0 +1,104 @@
+# Abstract
+
+This document outlines using the `Engine` to programmatically process data.
+
+# Overview
+
+The `Engine` is used to programmatically process data. There are three (3)
+major components to be aware of when using the `Engine`:
+- The `PluginSet`
+- The `DataSourceSet`
+- The `Engine` itself
+
+## Plugin Set
+
+The `PluginSet` is a collection of all of your plugins and extensions that can
+be used to process data from data sources (i.e. files.) You can create a new
+`PluginSet` by loading one or more extension directories:
+
+````cs
+using var plugins = PluginSet.Load("c:\\my\\plugin\\directory");
+````
+
+In advanced scenarios, overloads are provided to allow for you to pass your
+own `IAssemblyLoader` or `IPreloadValidator`. An example of this would be if
+you had an `IAssemblyLoader` implementation that loaded each plugin into its
+own isolated context.
+
+If you do not specify any directories, then the `PluginSet` will load all
+plugins that are found in the current working directory.
+
+## DataSourceSet
+
+The `DataSourceSet` is a collection of `IDataSource` instances that are to be
+processed by an `Engine` instance. The `DataSourceSet` will reference a
+`PluginSet` in order to validate that any `IDataSource` instances added to the
+`DataSourceSet` are able to be processed.
+
+````cs
+using var plugins = PluginSet.Load();
+using var dataSources = DataSourceSet.Create(plugins);
+
+dataSources.AddDataSource(new FileDataSource("myfile.txt"));
+dataSources.AddDataSource(new FileDataSource("yourfile.txt"));
+
+// ... and so on
+
+````
+
+If an attempt is made to add an `IDataSource` to the `DataSourceSet` for which
+no plugin can process, then the `DataSourceSet` will throw an exception. The
+`DataSourceSet` class provides Try versions of each of the methods (e.g.
+`TryAddDataSource`) that will not throw if the `IDataSource` does not have a
+corresponding plugin that can process the data source.
```

Comment:
...will not throw if the `IDataSource` does not have a plugin to process the data source.

is a little simpler. I got confused with the negations